PWG/EMCAL/EMCALtasks/AliEmcalRejectMCBackground.h

```diff
@@ -0,0 +1,186 @@
+   *
+   * @param outname name of the output contaienr
+   * @param nk Reject neutrons and K0long
+   * @param ch Select only charged particles
+   * @param etamax  Max eta acceptance
+   * @param physPrim Require physical primary particles
+   * @return AliEmcalRejectMCBackground*
+   */
+  static AliEmcalRejectMCBackground* AddTaskRejectMCBackground(TString outname = "mcparticlebgrej", Int_t signalRejection = 0, Int_t debug = 0);
+
+ protected:
+
+  /**
+   * @brief Creating user output
+   *
+   * Not used in this task
+   */
+  void UserCreateOutputObjects() {}
+
+  /**
+   * @brief Main event loop
+   *
+   * Run selection of particles and convert them to AliAODMCParticles and copy them
+   * to the output container. Set AliEmcalRejectMCBackground::AccpetParticle for the
+   * definition of selected particles.
+   *
+   * @param option Not used
+   */
+  void UserExec(Option_t *option);
+
+  /**
+   * @brief Convert MC particles in MC AOD articles (for ESD analysis).
+   * @param mcEvent Input event
+   * @param partOut Output particle container with selected particles
+   * @param partMap
+   */
+  void                      CreateParticleMap(AliVEvent *event, AliMCEvent* mcEvent, TClonesArray* partOut, AliNamedArrayI* partMap=0);
+
+  /**
+   * @brief Convert standard MC AOD particles in a new array, and filter if requested (for AOD analysis).
+   * @param partIn Input particle container
+   * @param partOut Output particle container with selected particles
+   * @param partMap Index map between particles in input and output container
+   */
+  void                      CreateParticleMapAOD(AliVEvent *event, AliMCEvent *mcEvent, TClonesArray* partIn, TClonesArray* partOut, AliNamedArrayI* partMap=0);
+
+  TString                   fParticlesOutName;     ///< name of output particle array
+  TString                   fParticlesMapName;     //!<! name of the particle map
+  Bool_t                    fInit;                 //!<! true = task initialized
+  TClonesArray             *fParticlesIn;          //!<! particle array in (AOD)
+  TClonesArray             *fParticlesOut;         //!<! particle array out
+  AliNamedArrayI           *fParticlesMap;         //!<! particle index/label
+  AliVEvent                *fEvent;                //!<! event
+  AliMCEvent               *fMC;                   //!<! MC event (ESD)
+  Bool_t                    fIsESD;                //!<! ESD or AOD analysis
+  Bool_t                    fDisabled;             //!<! Disable task if a problem occurs at initialization
+  Int_t                     fDebugLevel;           ///< debug level for interactive debugging
+  Int_t                     fnHeaders;             ///< Number of Headers
+  std::vector<int>          fNotRejectedStart;     //[fnHeaders]
+  std::vector<int>          fNotRejectedEnd;       //[fnHeaders]
+  std::vector<TString>      fGeneratorNames;       //[fnHeaders]
```

Comment:
Attention: having [] in comments is actually a ROOT extension for dynamic arrays reserving space (in this case of size fnHeaders) in the streamer. Since you are using a std::vector you don't make use of that, better remove, though without pointer it will probably be ignored by rootcling.